### DISCUSSION TITLE
Target 'mwpvlib' has 1 outputs: ['libmwpvlib.a'], but only 3 "install_dir"s were found.

### DISCUSSION THREAD

**Discussion by koklimabc** [stronnag/mwptools#174] [Q&A] [answered]
meson build --buildtype=release --strip --prefix /usr, output the errors as "Target 'mwpvlib' has 1 outputs: ['libmwpvlib.a'], but only 3 "install_dir"s were found."

How to fix it?
[meson-log.txt](https://github.com/stronnag/mwptools/files/15137340/meson-log.txt)

**Comment by stronnag** [comment]
Operating system and version?

**Comment by koklimabc** [comment]
ubuntu 23.10

**Comment by stronnag** [comment, marked as answer]
Thanks. I  have 22.04 and 24.04 here.
~~The reason is that Ubuntu 23.10 ships quite an old `meson` compared to the release of the OS.~~
The behaviour of `meson` changed in regard to the generation of libraries sometime between meson 1.2.0 and meson 1.3.0 (their release notes are good, bit not that granular). `mwp` checks for `1.2.0` to assess this change of behaviour; it is now apparent that `1.2.1` still has the old behaviour.

I've update the **development** branch on github?
Can you pull, checkout and build and see if that fixes the issue and let me know?
Ask here is you need help with getting `development` branch. 

Fixed in master 2024-04-27 (4966e9ff)

**Comment by koklimabc** [comment]
![Screenshot from 2024-04-27 16-13-46](https://github.com/stronnag/mwptools/assets/2577622/866bc7c2-be91-402b-8c44-2cc1c684efd4)

Yes already fix :), I build from source on ninja-1.12.0 version (then transfer it into '/usr/bin/ninja') and using meson 1.4.0 (BTW I do create separate environment (known as 'venv') to install latest meson version - pip install meson, refer 'https://askubuntu.com/questions/1465218/pip-error-on-ubuntu-externally-managed-environment-%C3%97-this-environment-is-extern'). 

meson build --buildtype=release --strip --prefix /usr
cd build
ninja install

And finally it work. Thanks very much to your reply :) .

**Comment by stronnag** [comment]
OK, but I already know that 1.4 works ... it would have been nice to confirm that bumping the version check fixed it on  23.10.
But never mind ...
